src/app/services/map.service.ts

```diff
@@ -0,0 +1,181 @@
+            idFieldName = idFieldName.replace(/\./g, '_');
+          }
+          return {
+            idFieldName: idFieldName,
+            idValue: id
+          };
+        });
+        this.mapComponent.selectFeaturesByCollection(featuresToSelect, mapContributor.collection);
+      } else {
+        this.mapComponent.selectFeaturesByCollection([], mapContributor.collection);
+      }
+    }
+  }
+
+  public getFeatureToHover(id: ElementIdentifier, mapContributor: MapContributor): FeatureHover {
+    const f = mapContributor.getFeatureToHightLight(id);
+    if (mapContributor) {
+      f.elementidentifier.idFieldName = f.elementidentifier.idFieldName.replace(/\./g, '_');
+    }
+    return f;
+  }
+
+  public setMapConfig(mapComponentConfig) {
+    this.mapComponentConfig = mapComponentConfig;
+  }
+
+  public getMapConfig() {
+    return this.mapComponentConfig;
+  }
+
+  public setCursor(cursor: string) {
+    this.mapComponent.map.getCanvas().style.cursor = cursor;
+  }
+
+  public setMapComponent(mapComponent: MapglComponent) {
+    this.mapComponent = mapComponent;
+  }
+
+  public clearWindowData(contributor: MapContributor) {
+    contributor.getConfigValue('layers_sources')
+      .filter(ls => ls.source.startsWith('feature-') && ls.render_mode === FeatureRenderMode.window)
+      .map(ls => ls.source)
+      .forEach(source => contributor.clearData(source));
+  }
+
+  public updateMapStyle(ids: Array<string | number>, collection: string) {
+    if (!!this.mapComponent && !!this.mapComponent.map && !!this.mapComponentConfig && !!this.mapComponentConfig.mapLayers.events.onHover) {
+      this.mapComponentConfig.mapLayers.events.onHover.forEach(l => {
+        const layer = this.mapComponent.map.getLayer(l) as ArlasAnyLayer;
+        if (!!layer && typeof(layer.source) === 'string' && layer.source.indexOf(collection) >= 0) {
+          if (ids && ids.length > 0) {
+            if (layer.metadata.isScrollableLayer || layer.metadata['is-scrollable-layer']) {
+              this.mapComponent.map.setFilter(l, this.getVisibleElementLayerFilter(l, ids));
+              const strokeLayerId = l.replace('_id:', '-fill_stroke-');
+              const strokeLayer = this.mapComponent.map.getLayer(strokeLayerId);
+              if (!!strokeLayer) {
+                this.mapComponent.map.setFilter(strokeLayerId, this.getVisibleElementLayerFilter(strokeLayerId, ids));
+              }
+            }
+          } else {
+            this.mapComponent.map.setFilter(l, this.mapComponent.layersMap.get(l).filter);
+            const strokeLayerId = l.replace('_id:', '-fill_stroke-');
+            const strokeLayer = this.mapComponent.map.getLayer(strokeLayerId);
+            if (!!strokeLayer) {
+              this.mapComponent.map.setFilter(strokeLayerId,
+                this.mapComponent.layersMap.get(strokeLayerId).filter);
+            }
+          }
+        }
+      });
+    }
+  }
+
+  public resize() {
+    this.mapComponent.map?.resize();
```

Comment:
check if mapComponent is undefined ?

Reply:
The resize method can be called whenever the window resizes anywhere from after the page initialisation because we listen to hat even in the constructor of the map and not in onMapLoaded for example. So it can happen before the map initialization, hence the ?

Reply:
same as above